pages/[resourceType]/create.tsx

```diff
@@ -0,0 +1,29 @@
+import { useRouter } from "next/router";
+import React from "react";
+import ResourceCodeEditor from "../../components/ResourceCodeEditor";
+import { Text } from "@mantine/core";
+/**
+ * NewResource is a page that renders a code editor for creating resources.
+ * @returns React node with a ResourceCodeEditor component
+ */
+const NewResource = () => {
```

Comment:
Nitpick, but can we rename this component? I think it's important to distinguish that this is a next page that renders the content for creating a new resource. I think just `NewResource` can be a little misleading.

How about `CreateResourcePage` or something?